docs/measurements/procedures/get_bad_bpms.md

```diff
@@ -0,0 +1,148 @@
+            #print(open(m).read())
+            bpm_tfs = tfs.read_tfs(m, index="NAME")
+            for bpm in bpm_tfs.index:
+                if bpm in appeared:
+                    continue
+                appeared.append(bpm)
+                if bpm in bad_bpms_list:
+                    bad_bpms_list[bpm] = bad_bpms_list[bpm] + 1
+                else:
+                    bad_bpms_list[bpm] = 1
+        return len(measurements_iforest)
+
+
+    def do_print(msg: str):
+        global OUTFILE
+        print(msg)
+        OUTFILE.write(f"{msg}\n")
+
+    def get_the_bad_bpms():
+        for plane in ["x", "y"]:
+            for beam in ["1", "2"]:
+
+                pl = "H" if plane == "x" else "V"
+                do_print("")
+                do_print(f"BEAM {beam} {pl}")
+                do_print("{:20} | {} | {} | {:8}  | {:7}".format("NAME", "PLANE", "BEAM", "FREQUENCY", "FROM"))
+                do_print("---------------------|-------|------|------------|-----------")
+                bad_bpms_list_iforest = {}
+                bad_bpms_list_svd = {}
+
+                n_svd = 0
+                n_iforest = 0
+                for date in DATES:
+                    print(f"date {date}")
+                    n_svd += get_bad_bpms_for_beam_and_plane_svd(date, plane, beam, bad_bpms_list_svd)
+                    n_iforest += get_bad_bpms_for_beam_and_plane_iforest(date, plane, beam, bad_bpms_list_iforest)
+
+                print(f"SVD [{n_svd} files]")
+                for (k,v) in bad_bpms_list_svd.items():
+                    v_perc = v / n_svd * 100
+                    if v_perc > 75:
+                        do_print(f"{k:20} | {plane:5} | {beam:4} | {v_perc:8.2f} % | SVD")
+                    elif v_perc > 50:
+                        do_print(f"{k:20} | {plane:5} | {beam:4} | {v_perc:8.2f} % | SVD")
+
+
+                print(f"IFOREST [{n_iforest} files]")
+                for (k,v) in bad_bpms_list_iforest.items():
+                    v_perc = v / n_iforest * 100
+                    if v_perc > 75:
+                        do_print(f"{k:20} | {plane:5} | {beam:4} | {v_perc:8.2f} % | IFOREST")
+                    elif v_perc > 50:
+                        do_print(f"{k:20} | {plane:5} | {beam:4} | {v_perc:8.2f} % | IFOREST")
+
+    if __name__ == "__main__":
+        main()
+
+    ```
+
+It scans all measurements in a list of given GUI output folders (`DATES`) and compiles a list of bad BPMs with their given number of appearances after SVD and isolation forest.
+Output will be written to `"bad_bpms.txt"`
+
+## Usage:
+- [ ] Define a set of representative measurements (those shoulf probably span different optics configurations and different analysis purposes)
+- [ ] Make sure that the measurements have the desired cleaning method applied.
+      If needed rerun the measurements with the GUI.
+- [ ] Adapt the `DATES` list at the beginning of this script accordingly
+- [ ] a) Run this script in `OP_DATA` or
```

Comment:
I'd rather say to run the script in the directory where GUI outputs are located, and specify that it can be `OP_DATA` if looking on the TN at the usual place.